src/hermes/commands/deposit/invenio.py

```diff
@@ -43,13 +49,81 @@ def map_metadata(ctx: CodeMetaContext):
     ctx.update(metadata_path, deposition_metadata)
 
 
-def _request_json(url: str) -> dict:
-    """Request an URL and return the JSON response as dict."""
+def deposit(click_ctx: click.Context, ctx: CodeMetaContext):
+    """Make a deposition on an Invenio-based platform.
+
+    This function can:
+
+    - Update the metadata of an existing record
+    - Update the metadata and files of an existing record y creating a new version
+    - Create a new record without any previous versions.
+    """
+
+    _log = logging.getLogger("cli.deposit.invenio")
 
-    # TODO: Store a requests.Session in a click_ctx in case we need it more frequently?
-    response = requests.get(url)
+    invenio_path = ContextPath.parse("deposit.invenio")
+    invenio_ctx = ctx[invenio_path]
+
+    if not click_ctx.auth_token:
+        raise DepositionUnauthorizedError("No auth token given for deposition platform")
+    click_ctx.session.headers["Authorization"] = f"Bearer {click_ctx.auth_token}"
+
+    # TODO: Get this from config or determine from some value (DOI, ...) in config.
+    existing_record_url = None
+
+    deposit_url = f"{invenio_ctx['siteUrl']}/{invenio_ctx['apiPaths']['depositions']}"
+    if existing_record_url is not None:
+        # TODO: Get by calling new version on existing record
+        deposit_url = None
+        raise NotImplementedError(
+            "At the moment, hermes can not create new versions of existing records"
+        )
+
+    deposition_metadata = invenio_ctx["depositionMetadata"]
+    response = click_ctx.session.post(
+        deposit_url,
+        json={"metadata": deposition_metadata}
+    )
     response.raise_for_status()
-    return response.json()
+
+    deposit = response.json()
+    _log.debug("Created deposit: %s", deposit["links"]["html"])
+
+    # ``deposit["metadata"]`` contains a ``prereserve_doi`` field which is a dict like
+    # this: ``{'doi': '10.5072/zenodo.1234567', 'recid': 1234567}``. The ``recid`` can
+    # be used to construct the URL to the record when published:
+    # ``f"https://sandbox.zenodo.org/record/{recid}"``.
+    # TODO: Use the URL and the DOI to update the data in the repo files before
+    # uploading them.
+
+    # TODO: When updating existing releases, the files API can be used to delete
+    # existing files. GET ``deposit["links"]["files"]`` for a list of ``files`` (names,
+    # checksums, ...), then DELETE ``file["links"]["download"]`` for each ``file`` in
+    # the list.
+
+    # Upload the files. We'll use the bucket API rather than the files API as it
+    # supports file sizes above 100MB.
+    bucket_url = deposit["links"]["bucket"]
+    file_name, file_content = _get_file_for_upload(deposition_metadata)
+
+    # TODO: When uploading real files, open them in binary mode and pass the handle to
+    # the data kwarg. For now we'll upload just this one file.
+    # TODO: Do we need to URL-encode the ``file_name`` or does this always work?
```

Comment:
**suggestion:** So many TODOs, maybe an issue tracker is a better place to keep track of them.

Reply:
I moved some stuff into issues.